setlocation.sh

```diff
@@ -1,29 +1,90 @@
 #!/bin/bash
 
-if [ -e /etc/location.conf ]
-then
-	LAT="$(grep latitude /etc/location.conf | cut -d\= -f2 | xargs)"
-	LON="$(grep longitude /etc/location.conf | cut -d\= -f2 | xargs)"
-	ELEV="$(grep elevation /etc/location.conf | cut -d\= -f2 | xargs)"
-fi
+LOCATION_DIR="/etc/location.conf.d"
+LOCATION_FILE="${LOCATION_DIR}/location.conf"
+ALL_LOCATIONS="${LOCATION_DIR}/*.conf"
 
-DATA=$(zenity --forms --title="Geographic location" \
-	--text="Set your geographic location" \
-	--separator=";" \
-	--add-entry="Latitude ($LAT)" \
-	--add-entry="Longitude ($LON)" \
-	--add-entry="Elevation ($ELEV)")
+function create {
+	if [ -e $LOCATION_FILE ]
+	then
+		LAT="$(grep latitude $LOCATION_FILE | cut -d\= -f2 | xargs)"
+		LON="$(grep longitude $LOCATION_FILE | cut -d\= -f2 | xargs)"
+		ELEV="$(grep elevation $LOCATION_FILE | cut -d\= -f2 | xargs)"
+		NAM="$(grep name $LOCATION_FILE | cut -d\= -f2 | xargs -0)"
+	fi
 
-LAT="$(echo $DATA | cut -d\; -f1)"
-LON="$(echo $DATA | cut -d\; -f2)"
-ELEV="$(echo $DATA | cut -d\; -f3)"
+	DATA=$(zenity --forms --title="Geographic location" \
+		--text="Set your geographic location" \
+		--separator=";" \
+		--add-entry="Latitude ($LAT)" \
+		--add-entry="Longitude ($LON)" \
+		--add-entry="Elevation ($ELEV)" \
+		--add-entry="Name ($NAM)")
 
-if [ "$LAT" != "" ] && [ "$LON" != "" ] && [ "$ELEV" != "" ]
-then
-	cat > /etc/location.conf << EOF
+	LAT="$(echo $DATA | cut -d\; -f1)"
+	LON="$(echo $DATA | cut -d\; -f2)"
+	ELEV="$(echo $DATA | cut -d\; -f3)"
+	NAM="$(echo $DATA | cut -d\; -f4)"
+
+	if [ "$LAT" != "" ] && [ "$LON" != "" ] && [ "$ELEV" != "" ]  && [ "$NAM" != "" ]
+	then
+		mkdir -p ${LOCATION_DIR}
```

Comment:
This will not work for unprivileged users i.e. /etc directory is owned by root with access rights set to drwxr-xr-x

Reply:
Actually my previous comment does not make sense, because setlocation.sh must be run as privileged user anyway.